R/FilterEnsemble.R

```diff
@@ -0,0 +1,105 @@
+
+
+#' @title Filter Ensemble
+#'
+#' @usage NULL
+#' @name mlr_filters_ensemble
+#' @format [`R6Class`][R6::R6Class] object inheriting from [`Filter`][mlr3filters::Filter].
+#'
+#' @description
+#' Implements the filte rensemble proposed in `r cite_bib("binder_2020")`.
+
+FilterEnsemble = R6Class("FilterEnsemble", inherit = mlr3filters::Filter,
+  public = list(
+    initialize = function(filters) {
+      private$.wrapped = assert_list(filters, types = "Filter", min.len = 1)
+      fnames = map_chr(private$.wrapped, "id")
+      names(private$.wrapped) = fnames
+      types_list = map(discard(private$.wrapped, function(x) test_scalar_na(x$task_types)), "task_types")
+      if (length(types_list)) {
+        task_types = Reduce(intersect, types_list)
+      } else {
+        task_types = NA_character_
+      }
+      .own_param_set = ps(
+        weights = p_uty(custom_check = crate(function(x) {
+          check_numeric(x, len = length(fnames)) %check&&%
+            (check_names(names(x), type = "unnamed") %check||%
+              check_names(names(x), type = "unique", permutation.of = fnames))
+          }, fnames),
+          tags = "required"
+        ),
+        rank_transform = p_lgl(init = FALSE, tags = "required")
+      )
+
+      super$initialize(
+        id = paste(fnames, collapse = "."),
+        task_types = task_types,
+        task_properties = unique(unlist(map(private$.wrapped, "task_properties"))),
+        param_set = ParamSetCollection$new(c(list(.own_param_set), map(private$.wrapped, "param_set"))),
```

Comment:
check the history of PipeOpLearner, before the PipeOp base class handled paramset collection / alist stuff properly. See there how cloning for paramset works. test this thoroughly.